src/trie.js

```diff
@@ -0,0 +1,53 @@
+const VALUE = Symbol('value');
+
+/**
+ * Created search trie
+ * @param {string[][]} lines
+ * @example
+ *  buildSearchTrie([
+ *   ['path1','value1'],
+ *   ['path1.path2','value2'],
+ *  ]);
+ */
+export const buildSearchTrie = (lines, separator = '/') => {
+    const root = {};
+    for (let i = 0; i < lines.length; ++i) {
+        const path = lines[i][0].split(separator);
+        let node = root;
+        // const lastIndex = path.length - 1;
+        for (let j = 0; j < path.length; ++j) {
+            const item = path[j];
+            if (!node[item]) {
+                node[item] = {};
+            }
+            node = node[item];
+        }
+        node[VALUE] = lines[i][1];
```

Comment:
@theKashey  how come it is `[i][1]` and not `[i][0]` here?

Reply:
Because we are storing `value` inside `VALUE`. The `key` is `[0]` and it was used to provide us with a current `node` value.